### ISSUE TITLE
Can't set vars_prompt default using a variable

### ISSUE THREAD

**Issue by jkarneges** [ansible/ansible#11757]
Hi folks, just started playing with Ansible. Great project!

I noticed I can't put a variable in the `default:` field of a prompted variable. Maybe it's a bug?

``` yaml
# test.yml

---
- hosts: localhost
  vars:
    test_var1: a nice default
  vars_prompt:
    - name: test_var2
      prompt: "Enter a value"
      default: "{{ test_var1 }}"
  tasks:
    - debug: var=test_var2
```

```
$ ansible-playbook test.yml
Enter a value [{{ test_var1 }}]:
```

I was expecting:

```
$ ansible-playbook test.yml
Enter a value [a nice default]:
```

**Comment by bcoca** [comment]
So this is not a feature we are going to implement as we need the vars themselves to be able to dynamically define other parts of the play and there is very little use to template it for the case shown above as you could just define default directly.

**Comment by jkarneges** [comment]
Bummer. For my use case I'm making a playbook that creates an account on a remote server, and I wanted the username to be configurable, but also default to the user's local username ($USER).

E.g.:

```
Username of account to create [jkarneges]:
```

This way, the user doesn't have to type anything nor use --extra-args, at least in the common case. But, if necessary, they get a chance to type something else.

**Comment by bcoca** [comment]
you can do this:

```
user: name={{promptvar|defaul(lookup('env', 'USER'))}} 
```

and leave the promptvar default to None

**Comment by kudos** [comment]
@bcoca what is promptvar? It doesn't appear in the codebase as far as I can tell.

**Comment by bcoca** [comment]
its a random name for a variable

**Comment by emcniece** [comment]
No idea how @bcoca is trying to implement this - some surrounding context would be useful, I can't manage to get it working. This would be a really useful feature!

**Comment by bdowling** [comment]
I came across this old issue attempting to do exactly the same.  It would still be nice to be able to template variable defaults based on other vars or facts.  e.g. a prior param could be input into a follow-on param such as specifying a top-level directory and having subdirs default to being in the same top-level or allowing the user to specify a whole different hierarchy.

I also see the potential for cases where you may wish to save previous values in a cache, load that in as the new defaults but still allow someone to change them.

In summary templated variable defaults are a useful feature.

**Comment by Celadora** [comment]
What you may want to do instead is pass your default variables using the --extra-vars flag and keep your prompts with basic defaults.

For example given the following playbook:
```

---
- hosts: 127.0.0.1
  connection: local

  vars_prompt:
  - name: "name"
    prompt: 'what is your name?'
    default: 'King Arthur'
    
  - name: "quest"
    prompt: "what is your quest?"
    default: 'To find the Holy Grail'
    
  - name: "favcolor"
    prompt: "what is your favorite color?"
    default: 'What do you mean? An African or European swallow?'


```

You could use the --extra-vars tag to explicitly set any of those variables, and the prompt would be skipped, thereby eliminating your need for a variable default.
`
$ ansible-playbook playbook.yml --extra-vars="name='Daffy Duck'"
`

**Comment by DanLipsitt** [comment]
I have a similar need. I want to get the latest tag for a repo using `github_release` and then use that value as a default but prompt the user to provide a different tag if desired. What is the recommended way to do this?
